src/App.tsx

```diff
@@ -1,24 +1,284 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect, useState, useMemo, useRef,
+} from 'react';
+import { Todo } from './types/Todo';
+import { TodosList } from './components/TodosList/TodosList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { Loader } from './components/Loader';
 
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { ERROR_MESSAGES } from './utils/constants/ERROR_MESSAGES';
+import { getFilteredTodos } from './utils/helpers/getFilteredTodos';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodos, setLoadingTodos] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectFilter, setSelectFilter] = useState(FilterType.All);
+  const textInputRef = useRef<HTMLInputElement | null>(null);
+  const [isProsessingTodoIds, setIsProsessingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    todoService.getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ERROR_MESSAGES.unableToLoadTodos);
+      })
+      .finally(() => {
+        setLoadingTodos(false);
+
+        textInputRef.current?.removeAttribute('disabled');
+        if (textInputRef.current) {
+          textInputRef.current.focus();
+        }
```

Comment:
https://github.com/mate-academy/react_todo-app-with-api/pull/824#discussion_r1355043809 - I see that you have created a function for this, but you don't use it everywhere